src/olmo_core/data/utils.py

```diff
@@ -419,8 +421,10 @@ def segment_documents_into_instances(
     Returns the number of original documents and the number of resulting instances documents.
     """
     total_og_docs = 0
-    indices = []
+    indices: List[int] = []
     for start_idx, end_idx in iter_document_indices(path, eos_token_id=eos_token_id, dtype=dtype):
+        if max_instances and len(indices) // 2 >= max_instances:
```

Comment:
Minor nit, but I think the semantics are more clear this way:
```suggestion
        if max_instances is not None and len(indices) // 2 >= max_instances:
```